File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/view/parameters/ProjectScreenData.java
package com.esofthead.mycollab.module.project.view.parameters;

import com.esofthead.mycollab.module.project.domain.Project;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

public class ProjectScreenData {
	public static class Goto extends ScreenData<Integer> {

		public Goto(Integer params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<Project> {

		public Add(Project params) {
			super(params);
		}
	}

	public static class Edit extends ScreenData<Project> {

		public Edit(Project params) {
			super(params);
		}
	}
}

File: mycollab-web/src/main/scala/com.esofthead.mycollab.module.project.view.parameters/ProjectScreenData.scala
package com.esofthead.mycollab.module.project.view.parameters

import com.esofthead.mycollab.module.project.domain.Project
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
object ProjectScreenData {

  class Goto(params: Integer) extends ScreenData[Integer](params) {}

  class Edit(params: Project) extends ScreenData[Project](params) {}

  class GotoTagList(params: Object) extends ScreenData[Object](params) {}

  class SearchItem(params: String) extends ScreenData[String](params) {}

  class GotoGanttChart extends ScreenData {}

  class GotoCalendarView extends ScreenData {}

}
